### PR TITLE
Feature/neo4j4 cluster

### PR DESCRIPTION
**Pull request by harshitsinghvi22**
Added support for versionless
Added support for Neo4j4
Added support for tagging virtual machine scale sets

### CODE REVIEW

File: scripts/node4.sh [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+azLoginIdentity=${12}
+resourceGroup=${13}
+vmScaleSetsName=${14}
+
+echo "Turning off firewalld"
+systemctl stop firewalld
+systemctl disable firewalld
+
+#Format and mount the data disk to /var/lib/neo4j
+MOUNT_POINT="/var/lib/neo4j"
```
Comment (ojhughes) [inline scripts/node4.sh:26]: can this be moved to a function

File: scripts/node4.sh [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+install_apoc_plugin() {
+  echo "Installing APOC..."
+  mv /var/lib/neo4j/labs/apoc-*-core.jar /var/lib/neo4j/plugins
+}
+
+get_latest_neo4j_version() {
+  echo "Getting latest neo4j version"
+  latest_neo4j_version=$(curl -s --fail http://versions.neo4j-templates.com/target.json | jq -r '.azure."4.4"' || echo "")
```
Comment (ojhughes) [inline scripts/node4.sh:86]: better to use `local -r` to scope variable to that function

File: scripts/node4.sh [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+  echo "Latest Neo4j Version is ${latest_neo4j_version}"
+}
+
+get_vmss_tags() {
+  taggedNeo4jVersion=$(az vmss list --resource-group ${resourceGroup} | jq --arg vmssName "${vmScaleSetsName}" '.[] | select(.name==$vmssName).tags.Neo4jVersion')
+  echo "Tagged Neo4j Version ${taggedNeo4jVersion}"
+}
+
+set_vmss_tags() {
+  installed_neo4j_version=$(/usr/bin/neo4j --version)
```
Comment (ojhughes) [inline scripts/node4.sh:96]: consider `local -r`

File: scripts/node4.sh [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+  echo "Installed neo4j version is ${installed_neo4j_version}. Trying to set vmss tags"
+  resourceId=$(az vmss list --resource-group "${resourceGroup}" | jq -r '.[] | .id')
+  az tag create --tags Neo4jVersion="${installed_neo4j_version}" --resource-id "${resourceId}"
+  echo "Added tag Neo4jVersion=${installed_neo4j_version}"
+}
+
+set_yum_pkg() {
+    yumPkg="neo4j-enterprise"
```
Comment (ojhughes) [inline scripts/node4.sh:104]: consider `local yumPkg`

File: scripts/node4.sh [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+  sed -i s/#dbms.routing.enabled=false/dbms.routing.enabled=true/g /etc/neo4j/neo4j.conf
+  sed -i s/#dbms.routing.advertised_address=:7688/dbms.routing.advertised_address=${privateIP}:7688/g /etc/neo4j/neo4j.conf
+  sed -i s/#dbms.routing.listen_address=0.0.0.0:7688/dbms.routing.listen_address=${privateIP}:7688/g /etc/neo4j/neo4j.conf
+  echo dbms.routing.default_router=SERVER >> /etc/neo4j/neo4j.conf
+}
+
+build_neo4j_conf_file() {
+  echo "Configuring network in neo4j.conf..."
+
+  nodeIndex=`curl -H Metadata:true "http://169.254.169.254/metadata/instance/compute?api-version=2017-03-01" \
```
Comment (ojhughes) [inline scripts/node4.sh:181]: consider `local -r`

File: scripts/node4.sh (hunk 4 above)
Comment (harshitsinghvi22) [reply scripts/node4.sh:104]: this is required outside the func hence cannot be local

File: scripts/node4.sh (hunk 3 above)
Comment (harshitsinghvi22) [reply scripts/node4.sh:96]: this is required outside the func hence cannot be local

File: scripts/node4.sh (hunk 2 above)
Comment (harshitsinghvi22) [reply scripts/node4.sh:86]: this is required outside the func hence cannot be local

File: scripts/node4.sh (hunk 1 above)
Comment (harshitsinghvi22) [reply scripts/node4.sh:26]: this is done

File: scripts/node4.sh [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+graphDataScienceLicenseKey=$7
+installBloom=$8
+bloomLicenseKey=$9
+nodeCount=${10}
+loadBalancerDNSName=${11}
+azLoginIdentity=${12}
+resourceGroup=${13}
+vmScaleSetsName=${14}
+
+echo "Turning off firewalld"
```
[the remark this answers is not in the document]
Comment (harshitsinghvi22) [reply scripts/node4.sh:21]: will make this as a separate PR ...will need to see how this works out in azure

File: scripts/node4.sh [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+systemctl daemon-reload
+mount -a
+
+
+install_azure_from_dnf() {
+
+  sudo rpm --import https://packages.microsoft.com/keys/microsoft.asc
+  sudo dnf install -y https://packages.microsoft.com/config/rhel/8/packages-microsoft-prod.rpm
+echo -e "[azure-cli]
```
[the remark this answers is not in the document]
Comment (harshitsinghvi22) [reply scripts/node4.sh:47]: not a big fan of heredoc to be honest...echo is pretty simple to read but have added the change as requested

File: scripts/node4.sh (hunk 5 above)
Comment (harshitsinghvi22) [reply scripts/node4.sh:181]: done

**Review by ojhughes — APPROVED** [review]
Great stuff thanks

**Review by bfeshti — APPROVED** [review]

**Pull request merged**